tests/benchmarks/android_jankbench.py

```diff
@@ -0,0 +1,196 @@
+            rate_limit_us = 2000
+            # Different schedutil versions have different tunables
+            tunables = self.target.cpufreq.list_governor_tunables(0)
+            if 'rate_limit_us' in tunables:
+                tunables = {'rate_limit_us' : str(rate_limit_us)}
+            else:
+                assert ('up_rate_limit_us' in tunables and
+                        'down_rate_limit_us' in tunables)
+                tunables = {
+                    'up_rate_limit_us' : str(rate_limit_us),
+                    'down_rate_limit_us' : str(rate_limit_us)
+                }
+
+            try:
+                for cpu_id in range(self.te.platform['cpus_count']):
+                    self.target.cpufreq.set_governor_tunables(
+                        cpu_id, 'schedutil', **tunables)
+            except TargetError as e:
+                self._log.warning('Failed to set schedutils parameters: {}'\
+                                 .format(e))
+                raise
+            self._log.info('Set schedutil.rate_limit_us=%d', rate_limit_us)
+
+        # Setup ondemand parameters
+        if self.governor == 'ondemand':
+            try:
+                for cpu_id in range(self.te.platform['cpus_count']):
+                    tunables = self.target.cpufreq.get_governor_tunables(cpu_id)
+                    self.target.cpufreq.set_governor_tunables(
+                        cpu_id, 'ondemand',
+                        **{'sampling_rate' : tunables['sampling_rate_min']})
+            except TargetError as e:
+                self._log.warning('Failed to set ondemand parameters: {}'\
+                                 .format(e))
+                raise
+            self._log.info('Set ondemand.sampling_rate to minimum supported')
+
+        # Report configured governor
+        governors = self.target.cpufreq.get_all_governors()
+        self._log.info('Using governors: %s', governors)
+
+
+# Run the benchmark in each of the supported governors
+
+iterations = 1
+
+governors = [
+    'performance',
+    'powersave',
+    'ondemand',
+    'interactive',
+    'sched',
+    'schedutil'
+]
+
+tests = [
+    'list_view',
+    'image_list_view',
+    'shadow_grid',
+    'low_hitrate_text',
+    'high_hitrate_text',
+    'edit_text'
+]
+
+# Reboot device only the first time
+do_reboot = True
+tests_remaining = len(governors) * len(tests)
+tests_completed = 0
+for governor in governors:
+    for test in tests:
+        tests_remaining -= 1
+        delay_after_s = 30 if tests_remaining else 0
+        try:
+            JankbenchTest(governor, test, iterations,
+                          do_reboot, delay_after_s)
+            tests_completed += 1
+        except:
+            # A test configuraion failed, continue with other tests
+            pass
+        do_reboot = False
+
+# We want to collect data from at least two different governors
+# to have sensible data to compare from this test.
+assert(tests_completed >= 2)
```

Comment:
I think it might be best if we assert that we were able to run for at least 1 governor, and not 2. Because we might want run a test for a kernel with one governor and compare with the results for another kernel tested with the same governor. I believe the failure conditions should only be if none of the tests were able to successfully complete.